File: ONLINE_QUALIFIERS_ROUND_2/7_PO.cpp
#include <iostream>
#include <vector>
#include <algorithm>
using namespace std;
int minimizeProjectTime(int n, vector<int> &mainTime, vector<int> &altTime,
                        vector<int> &switchToAlt, vector<int> &switchToMain,
                        int mainStart, int altStart, int mainDeploy, int altDeploy)
{

    vector<int> dpMain(n), dpAlt(n);
    dpMain[0] = mainStart + mainTime[0];
    dpAlt[0] = altStart + altTime[0];
    for (int i = 1; i < n; ++i)
    {
        dpMain[i] = min(
            dpMain[i - 1] + mainTime[i],                 
            dpAlt[i - 1] + switchToMain[i] + mainTime[i] 
        );

        dpAlt[i] = min(
            dpAlt[i - 1] + altTime[i],                  
            dpMain[i - 1] + switchToAlt[i] + altTime[i] 
        );
    }
    return min(dpMain[n - 1] + mainDeploy, dpAlt[n - 1] + altDeploy);
}
int main()
{
    int n;
    cin >> n;

    vector<int> mainTime(n), altTime(n);
    vector<int> switchToAlt(n), switchToMain(n);

    for (int i = 0; i < n; ++i)
        cin >> mainTime[i];
    for (int i = 0; i < n; ++i)
        cin >> altTime[i];
    for (int i = 0; i < n; ++i)
        cin >> switchToAlt[i];
    for (int i = 0; i < n; ++i)
        cin >> switchToMain[i];

    int mainStart, altStart, mainDeploy, altDeploy;
    cin >> mainStart >> altStart >> mainDeploy >> altDeploy;

    int result = minimizeProjectTime(n, mainTime, altTime,
                                     switchToAlt, switchToMain,
                                     mainStart, altStart, mainDeploy, altDeploy);
    cout << result << endl;
    return 0;
}

File: MOCK_ROUND_2/7_PO.py
def minimize_project_time(n, m, a, tm, ta, m0, a0, dm, da):
    dp_main = [0] * n
    dp_alt = [0] * n

    dp_main[0] = m0 + m[0]
    dp_alt[0] = a0 + a[0]

    for i in range(1, n):
        dp_main[i] = min(
            dp_main[i - 1] + m[i],
            dp_alt[i - 1] + ta[i] + m[i], 
        )

        dp_alt[i] = min(
            dp_alt[i - 1] + a[i],
            dp_main[i - 1] + tm[i] + a[i],
        )

    result = min(dp_main[n - 1] + dm, dp_alt[n - 1] + da)
    return result
n = int(input())
m = list(map(int, input().split()))
a = list(map(int, input().split()))
tm = list(map(int, input().split()))
ta = list(map(int, input().split()))
m0, a0, dm, da = map(int, input().split())
print(minimize_project_time(n, m, a, tm, ta, m0, a0, dm, da))
